src/components/form/numeric-input.tsx

```diff
@@ -0,0 +1,59 @@
+import { IconDefinition } from "@fortawesome/fontawesome-common-types";
+import { faMinus, faPlus } from "@fortawesome/free-solid-svg-icons";
+import cx from "classnames";
+import { HTMLProps, forwardRef, useState } from "react";
+import { UseFormRegisterReturn } from "react-hook-form";
+import { Button } from "../button";
+
+type Props = UseFormRegisterReturn<string> &
+  HTMLProps<HTMLInputElement> & {
+    labelText?: string;
+    hintText?: string;
+    feedback?: string;
+    className?: string;
+    defaultValue?: number;
+  };
+
+export const NumericInput = forwardRef<HTMLInputElement, Props>(
+  ({ labelText, hintText, feedback, className, defaultValue, ...rest }: Props, ref) => {
+    const [value, setValue] = useState<number | undefined>(defaultValue);
+
+    return (
+      <div className="flex flex-col gap-1 leading-5 text-base font-normal font-text">
+        <label>{labelText}</label>
+        <input
+          {...rest}
+          ref={ref}
+          value={value}
+          onChange={(e) => {
+            const numericValue = e.target.value.replace(/[^0-9]/g, "");
+            setValue(Number(numericValue));
+          }}
+          className={cx(
+            `rounded-lg border border-black py-5 px-3 relative`,
+            className,
+          )}
+        />
+        <Button
+          variant="icon"
+          className="absolute top-14 right-20 z-1 border border-black"
+          icon={faMinus as IconDefinition}
+          onClick={() => setValue((prev) => (prev || 0) - 1)}
+        />
+        <Button
+          variant="icon"
+          className="absolute top-14 right-10 z-1 border border-black"
+          icon={faPlus as IconDefinition}
+          onClick={() => setValue((prev) => (prev || 0) + 1)}
```

Comment:
If we define `value` as a number instead of a string, these functions can use the previous value to set the next value and we can delete the onClick functions you had here before 👍